docs/src/pages/components/data-grid/rendering/RenderExpandCellGrid.tsx

```diff
@@ -0,0 +1,195 @@
+import * as React from 'react';
+import { Typography } from '@material-ui/core';
+import Paper from '@material-ui/core/Paper';
+import Popper from '@material-ui/core/Popper';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import {
+  DataGrid,
+  GridColDef,
+  GridCellParams,
+  isOverflown,
+} from '@material-ui/data-grid';
+
+interface GridCellExpandProps {
+  value: string;
+  width: number;
+}
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    root: {
+      alignItems: 'center',
+      lineHeight: '24px',
+      width: '100%',
+      height: '100%',
+      position: 'relative',
+      display: 'flex',
+      '& .MuiRating-root': {
+        marginRight: theme.spacing(1),
+      },
+      '& .cellValue': {
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      },
+    },
+  }),
+);
+
+const GridCellExpand = React.memo(function GridCellExpand(
+  props: GridCellExpandProps,
+) {
+  const { width, value } = props;
+  const wrapper = React.useRef<HTMLDivElement | null>(null);
+  const cellDiv = React.useRef(null);
+  const cellValue = React.useRef(null);
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const classes = useStyles();
+  const [showFullCell, setShowFullCell] = React.useState(false);
+  const [showPopper, setShowPopper] = React.useState(false);
+
+  const handleMouseEnter = () => {
+    const isCurrentlyOverflown = isOverflown(cellValue.current!);
+    setShowPopper(isCurrentlyOverflown);
+    setAnchorEl(cellDiv.current);
+    setShowFullCell(true);
+  };
+
+  const handleMouseLeave = () => {
+    setShowFullCell(false);
+  };
+
+  React.useEffect(() => {
+    if (!showFullCell) {
+      return undefined;
+    }
+
+    function handleKeyDown(nativeEvent: KeyboardEvent) {
+      // IE11, Edge (prior to using Bink?) use 'Esc'
+      if (nativeEvent.key === 'Escape' || nativeEvent.key === 'Esc') {
+        setShowFullCell(false);
+      }
+    }
+
+    document.addEventListener('keydown', handleKeyDown);
+
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [setShowFullCell, showFullCell]);
+
+  return (
+    <div
+      ref={wrapper}
+      className={classes.root}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <div
+        ref={cellDiv}
+        style={{
+          height: 1,
+          width,
+          display: 'block',
+          position: 'absolute',
+          top: 0,
+        }}
+      />
+      <div ref={cellValue} className="cellValue">
+        {value}
+      </div>
+      {showPopper && (
+        <Popper
+          open={showFullCell && anchorEl != null}
```

Comment:
It seems that it can't be undefined

```suggestion
          open={showFullCell && anchorEl !== null}
```